components/backstage/backstage_controller.jsx

```diff
@@ -39,69 +38,56 @@ const BackstageRoute = ({component: Component, extraProps, ...rest}) => ( //esli
 );
 
 export default class BackstageController extends React.Component {
-    constructor(props) {
-        super(props);
+    static propTypes = {
 
-        this.onTeamChange = this.onTeamChange.bind(this);
-        this.onUserChange = this.onUserChange.bind(this);
+        /**
+         * Current user.
+         */
+        user: PropTypes.object,
 
-        const team = TeamStore.getCurrent();
-        const user = UserStore.getCurrentUser();
+        /**
+         * Current team.
+         */
+        team: PropTypes.object,
 
-        this.state = {
-            user,
-            team,
-            isAdmin: UserStore.isSystemAdminForCurrentUser(user) ||
-                TeamStore.isTeamAdminForCurrentTeam(team)
-        };
-    }
-
-    componentDidMount() {
-        TeamStore.addChangeListener(this.onTeamChange);
-        UserStore.addChangeListener(this.onUserChange);
-    }
-
-    componentWillUnmount() {
-        TeamStore.removeChangeListener(this.onTeamChange);
-        UserStore.removeChangeListener(this.onUserChange);
+        /**
+         * Set to indicate user is system admin or a team admin for current team.
+         */
+        isAdmin: PropTypes.bool
     }
 
-    onUserChange() {
-        const user = UserStore.getCurrentUser();
-
-        this.setState({
-            user
-        });
+    scrollToTop = () => {
+        if (this.listRef) {
+            this.listRef.scrollTop = 0;
+        }
     }
 
-    onTeamChange() {
-        const team = TeamStore.getCurrent();
-
-        this.setState({
-            team,
-            isAdmin: UserStore.isSystemAdminForCurrentUser(this.state.user) ||
-                TeamStore.isTeamAdminForCurrentTeam(team)
-        });
+    setListRef = (ref) => {
+        this.listRef = ref;
     }
 
     render() {
-        if (this.state.team == null || this.state.user == null) {
+        if (this.props.team == null || this.props.user == null) {
```

Comment:
I guess this confirms that props `team` and `user` are actually optional. Being extra-vigilant as per last dev meeting. It would be cool if the `PropTypes` API were `.optional` rather than `.required`, come to think of it.